free-3.md

```diff
@@ -10,34 +11,49 @@ _What need is met by your idea? Who are you helping? What is that person's
+
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+The primary benefit of a DSL in this situation would be to simplify bot creation. It addresses the need for a tool that is usable by people who do not have programming knowledge. The DSL would basically be a graphical overlay for bot creation APIs.
+
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I have tried to create bots for social media in the past and ran into various errors throughout the process. Even as someone with a computer science background, it is difficult to get a bot running. I have seen a wide variety of novelty bots ranging from clever to annoying. My idea came from the desire to see more novelty bots without all of the pain involved in creating a custom bot.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Easier social media bot creation.
+
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+I could imagine the user choosing a site and then naming their bot. They would have to go through some basic set up information in a GUI and then would have the means to define the functionality of their bot. The most basic functionality for the bot could be to post something specific in response to a key phrase or to make random posts selected from a certain data set. In the interest of starting small and gradually growing the language, the user could select from a set number of bot types and add options to customize their bot. A program would then be the framework to create the bot, including the website, the setup information for the bot, and the basic behavior of the bot. This is a simpler (if not more limited) way to interact with the problem domain, which is suitable for the targeted user.
+
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When a program runs, the bot would be created and it would begin to operate. Since this is a long and involved process, it could be better to split the program between bot creation and operation. The user could get errors if the bot was not created, which could be translated from the website's error messages into simple instructions for the user. Other errors that could occur involve the bot not functioning properly within the domain of the website (e.g. trying to post things longer than 140 characters on Twitter). This could be detected while the user "programs" the bot and display error messages. The developer would also need to make it easy for the user to run their bot, which would involve keeping it running on their computer or figuring out another solution with few complications for the user.
```

Comment:
I like the idea of splitting the program between bot creation and operation. Perhaps even focusing on only one aspect to make it a feasible project to be completed this semester.